### DISCUSSION TITLE
Pro: Reset current snap when shortcut/modifier is released?

### DISCUSSION THREAD

**Discussion by deitrix** [rxhanson/RectanglePro-Community#226] [Q&A]
I'm using the Pro version and I was wondering if there's a way to "forget" the current execution when using custom shortcuts?

I've bound (some modifier) + 2 to centre my windows, and subsequent keypresses will cycle between different size configurations (executions) in the centre of the screen. My issue is that once I go to snap a window that's already in one of the centre snap configurations, it "remembers" which execution it's in and cycles to the next one in the list. I wish for it to start from the first execution again. Whenever my keyboard modifier/shortcut is released, it should "forget" its progression through the list of executions.

Is this possible with the current version, or would it require a code change?

**Comment by rxhanson** [comment]
Thanks for using Pro. Is this what you're looking for?

![image](https://user-images.githubusercontent.com/13651296/219441482-35239ffe-e1e7-457d-af6f-d2e032020d89.png)

**Comment by deitrix** [comment]
This is the section I'm referring to:

![image](https://user-images.githubusercontent.com/14089691/219443950-a39ea43c-623d-462e-87b0-da8f1d17aa04.png)

My issue is that when I repeatedly press the shortcut for a particular configuration, when I come back to resize the window later, it remembers the execution that its in, and cycles through the executions from there.

Say for example I press the shortcut for "Fill Center" twice, putting it in the "x 2/8, y0 w 3/8, h1" configuration. Once I'm finished doing whatever task requires that configuration, I want to return to the primary configuration (the first one) of "x 2/8, y 0 w 4/8, h 1". To do this, I have to press the shortcut 3 times, when I'd prefer if I only had to press it once.

My executions are ordered by how often I use that particular configuration. So the first execution is always the most commonly used. If I had lots of executions for a shortcut, it would take forever to get a window back to its primary configuration. Does that make any sense?

**Reply by rxhanson** [reply]
I see. Would you prefer something like a timeout reset, or a reset when a different window is managed?

**Reply by deitrix** [reply]
A timeout reset would by acceptable.

Ideally, it would reset when I lift off of my modifier keys, but I get that you're probably not distinguishing between modifiers and the _rest_ of the shortcut. For instance, right now I have it set up as cmd+ctrl+1, cmd+ctrl+2 and cmd+ctrl+3. cmd+ctrl being the modifier combination and 1, 2 and 3 being the keys I press to trigger the custom shortcuts. Rectangle should only remember the last shortcut execution for the time that cmd+ctrl are pressed down. As soon as I lift off of cmd+ctrl, it forgets the current execution/snap state for any of the shortcuts.

One workaround for the time being is that I can quickly trigger one of the other shortcuts, moving the window temporarily, and then triggering the shortcut that I actually want. That puts it into the first execution state for that shortcut.

All that being said, a timeout reset would solve my problem for 99% of scenarios. Also, thanks for the quick reply 😃

**Reply by rxhanson** [reply]
I'll see if I can fit something into an upcoming release for it, perhaps as just a Terminal command configuration to make it something I can get out the door fast.

**Reply by deitrix** [reply]
Thank you, I really appreciate it! 😊

**Reply by rxhanson** [reply]
In the latest version, I added a checkbox for this:

![image](https://user-images.githubusercontent.com/13651296/220709394-5147f50c-eba7-4164-a2f2-f373d50ce6b7.png)

It applies to all repetitions, not just custom ones. Let me know if works how you expected.

**Reply by deitrix** [reply]
That works perfectly. Thank you! 😄
